pkg/compare/compare.go

```diff
@@ -146,6 +151,59 @@ func UnmarshalInlineYaml(obj map[string]interface{}, targetPath string) (err err
 	return
 }
 
+// genPathIgnoreOpt returns a cmp.Option to ignore paths specified in parameter ignorePaths.
+func genYamlIgnoreOpt(yamlStr string) (cmp.Option, error) {
+	tree := make(map[string]interface{})
+	if err := yaml.Unmarshal([]byte(yamlStr), &tree); err != nil {
+		return nil, err
+	}
+	return cmp.FilterPath(func(curPath cmp.Path) bool {
+		up := pathToStringList(curPath)
+		found, _ := IsPathInTree(tree, up)
+		return found
+	}, cmp.Ignore()), nil
+}
+
+func IsPathInTree(tree map[string]interface{}, path []string) (bool, error) {
+	// Clone a slice: https://github.com/go101/go101/wiki
+	remainPath := append(path[:0:0], path...)
+	return isPathInTree(tree, remainPath)
+}
+
+func isPathInTree(tree map[string]interface{}, remainPath []string) (bool, error) {
+	if len(remainPath) == 0 {
+		return false, nil
+	}
+	for key, val := range tree {
```

Comment:
could you repurpose one of the existing tree traversals (e.g. read node) rather than copying the code here?

Reply:
Because the potential impact of repurposing tree traversal func is big. We need to do that in a separate round of refactoring and we already had an issue to track it: https://github.com/istio/istio/issues/15494.